### PR TITLE
[CPU] Code generation: Floor & Ceiling & Round implementation

### PR DESCRIPTION
**Pull request by eshoguli**
### Details:
 - *Code generation: Floor & Ceiling & Round implementation*

### Tickets:
 - *71015*
 - *PR with tests - WIP*

### CODE REVIEW

**Review by tgubanova-lohika — CHANGES_REQUESTED** [review]
Stop russian aggression in Ukraine!

File: src/plugins/intel_cpu/src/emitters/jit_eltwise_emitters.cpp [hunk 1]
```diff
@@ -258,6 +258,239 @@ size_t jit_divide_emitter::aux_vecs_count() const {
...
+template <mkldnn::impl::cpu::x64::cpu_isa_t isa>
+void jit_floor_emitter::emit_isa(const std::vector<size_t> &in_vec_idxs, const std::vector<size_t> &out_vec_idxs) const {
+    using Vmm = typename conditional3<isa == cpu::x64::sse41, Xmm, isa == cpu::x64::avx2, Ymm, Zmm>::type;
+    Vmm vmm_src = Vmm(in_vec_idxs[0]);
+    Vmm vmm_dst = Vmm(out_vec_idxs[0]);
+    h->uni_vroundps(vmm_dst, vmm_src, 1);
+}
+
+size_t jit_floor_emitter::aux_vecs_count() const {
+    return 1;
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply src/plugins/intel_cpu/src/emitters/jit_eltwise_emitters.cpp:292]: done, thanks

File: src/plugins/intel_cpu/src/emitters/jit_eltwise_emitters.cpp [hunk 2]
```diff
@@ -258,6 +258,239 @@ size_t jit_divide_emitter::aux_vecs_count() const {
...
+}
+
+/// CEILING ///
+jit_ceiling_emitter::jit_ceiling_emitter(jit_generator *host, cpu_isa_t host_isa, const std::shared_ptr<ngraph::Node>& node, Precision exec_prc)
+: jit_emitter(host, host_isa, node, exec_prc) {}
+jit_ceiling_emitter::jit_ceiling_emitter(jit_generator* host, cpu_isa_t host_isa, Precision exec_prc)
+    : jit_emitter(host, host_isa, exec_prc) {}
+
+size_t jit_ceiling_emitter::get_inputs_num() const { return 2; }
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply src/plugins/intel_cpu/src/emitters/jit_eltwise_emitters.cpp:301]: done, thanks

File: src/plugins/intel_cpu/src/emitters/jit_eltwise_emitters.cpp [hunk 3]
```diff
@@ -258,6 +258,239 @@ size_t jit_divide_emitter::aux_vecs_count() const {
...
+        assert(!"unsupported isa");
+    }
+}
+
+template <mkldnn::impl::cpu::x64::cpu_isa_t isa>
+void jit_ceiling_emitter::emit_isa(const std::vector<size_t> &in_vec_idxs, const std::vector<size_t> &out_vec_idxs) const {
+    using Vmm = typename conditional3<isa == cpu::x64::sse41, Xmm, isa == cpu::x64::avx2, Ymm, Zmm>::type;
+    Vmm vmm_src = Vmm(in_vec_idxs[0]);
+    Vmm vmm_dst = Vmm(out_vec_idxs[0]);
+    h->uni_vroundps(vmm_dst, vmm_src, 2);
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply src/plugins/intel_cpu/src/emitters/jit_eltwise_emitters.cpp:320]: done, thanks.

File: src/plugins/intel_cpu/src/emitters/jit_eltwise_emitters.cpp [hunk 4]
```diff
@@ -258,6 +259,119 @@ size_t jit_divide_emitter::aux_vecs_count() const {
...
+}
+
+template <mkldnn::impl::cpu::x64::cpu_isa_t isa>
+void jit_round_emitter::emit_isa(const std::vector<size_t> &in_vec_idxs, const std::vector<size_t> &out_vec_idxs) const {
+    using Vmm = typename conditional3<isa == cpu::x64::sse41, Xmm, isa == cpu::x64::avx2, Ymm, Zmm>::type;
+    Vmm vmm_src = Vmm(in_vec_idxs[0]);
+    Vmm vmm_dst = Vmm(out_vec_idxs[0]);
+    // Round down: 01B - Rounded result is closest to but no greater than the infinitely precise result.
+    // Round up: 10B - Rounded result is closest to but no less than the infinitely precise result.
+    h->uni_vroundps(vmm_dst, vmm_src, mode == RoundMode::HALF_AWAY_FROM_ZERO ? 1 : 2);
```
Comment (a-sidorova) [inline src/plugins/intel_cpu/src/emitters/jit_eltwise_emitters.cpp:372]: It's incorrect for these modes.  You should reuse oneDNN emitters for these mods:  [HALF_AWAY_FROM_ZERO](https://github.com/openvinotoolkit/oneDNN/blob/82ca2f931c1d588b67d154d873136d4af1ffb3a8/src/cpu/x64/injectors/jit_uni_eltwise_injector.cpp#L1557) and [HALF_TO_EVEN](https://github.com/openvinotoolkit/oneDNN/blob/82ca2f931c1d588b67d154d873136d4af1ffb3a8/src/cpu/x64/injectors/jit_uni_eltwise_injector.cpp#L1551) as [Gelu-7](https://github.com/openvinotoolkit/openvino/blob/4b29eed013e1e86bab079831b5be7f126890daba/src/plugins/intel_cpu/src/emitters/jit_mkldnn_ext_emitters.hpp#L127) for example.

File: src/common/snippets/src/pass/collapse_subgraph.cpp [hunk 5]
```diff
@@ -9,6 +9,7 @@
 #include "snippets/op/subgraph.hpp"
```
Comment (a-sidorova) [inline src/common/snippets/src/pass/collapse_subgraph.cpp:9]: just random place: could you please add these ops [here](https://github.com/openvinotoolkit/openvino/blob/master/src/common/snippets/include/snippets/snippets_isa_tbl.hpp)?

File: src/plugins/intel_cpu/src/emitters/jit_eltwise_emitters.cpp (hunk 4 above)
Comment (eshoguli) [reply src/plugins/intel_cpu/src/emitters/jit_eltwise_emitters.cpp]: done, thanks

File: src/common/snippets/src/pass/collapse_subgraph.cpp (hunk 5 above)
Comment (eshoguli) [reply src/common/snippets/src/pass/collapse_subgraph.cpp:9]: done, thanks. But full type names (with namespaces) are used in snippets, looks like it's not used.

**Review by a-sidorova — APPROVED** [review]

File: src/plugins/intel_cpu/src/emitters/jit_mkldnn_ext_emitters.hpp [hunk 6]
```diff
@@ -142,5 +143,26 @@ class jit_gelu_v7_emitter : public jit_mkldnn_emitter {
...
+public:
+    jit_round_emitter(
+        mkldnn::impl::cpu::x64::jit_generator *host,
+        mkldnn::impl::cpu::x64::cpu_isa_t host_isa,
+        const std::shared_ptr<ngraph::Node>& n,
+        InferenceEngine::Precision exec_prc = InferenceEngine::Precision::FP32) : jit_mkldnn_emitter(host, host_isa, n, exec_prc) {
+        const auto round = getNgraphOpAs<ngraph::op::v5::Round>(n);
+        const auto node_mode = round->get_mode();
+        if ((node_mode != ngraph::opset5::Round::RoundMode::HALF_AWAY_FROM_ZERO &&
+             node_mode == ngraph::opset5::Round::RoundMode::HALF_TO_EVEN)) {
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply src/plugins/intel_cpu/src/emitters/jit_mkldnn_ext_emitters.hpp]: thanks! : ) done

File: src/plugins/intel_cpu/src/emitters/jit_mkldnn_ext_emitters.hpp [hunk 7]
```diff
@@ -142,5 +143,26 @@ class jit_gelu_v7_emitter : public jit_mkldnn_emitter {
...
+    jit_round_emitter(
+        mkldnn::impl::cpu::x64::jit_generator *host,
+        mkldnn::impl::cpu::x64::cpu_isa_t host_isa,
+        const std::shared_ptr<ngraph::Node>& n,
+        InferenceEngine::Precision exec_prc = InferenceEngine::Precision::FP32) : jit_mkldnn_emitter(host, host_isa, n, exec_prc) {
+        const auto round = getNgraphOpAs<ngraph::op::v5::Round>(n);
+        const auto node_mode = round->get_mode();
+        if ((node_mode != ngraph::opset5::Round::RoundMode::HALF_AWAY_FROM_ZERO) &&
+             (node_mode != ngraph::opset5::Round::RoundMode::HALF_TO_EVEN)) {
+            IE_THROW(NotImplemented) << "Subgraph node doesn't support ngraph operation Round with mode: " << static_cast<int>(node_mode);
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/emitters/jit_mkldnn_ext_emitters.hpp:157]: `Subgraph node` -> `Round emitter`

File: src/plugins/intel_cpu/src/emitters/jit_mkldnn_ext_emitters.hpp [hunk 8]
```diff
@@ -142,5 +143,26 @@ class jit_gelu_v7_emitter : public jit_mkldnn_emitter {
...
 
+class jit_round_emitter : public jit_mkldnn_emitter {
+public:
+    jit_round_emitter(
+        mkldnn::impl::cpu::x64::jit_generator *host,
+        mkldnn::impl::cpu::x64::cpu_isa_t host_isa,
+        const std::shared_ptr<ngraph::Node>& n,
+        InferenceEngine::Precision exec_prc = InferenceEngine::Precision::FP32) : jit_mkldnn_emitter(host, host_isa, n, exec_prc) {
+        const auto round = getNgraphOpAs<ngraph::op::v5::Round>(n);
+        const auto node_mode = round->get_mode();
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply src/plugins/intel_cpu/src/emitters/jit_mkldnn_ext_emitters.hpp]: done, thanks

File: src/plugins/intel_cpu/src/emitters/jit_mkldnn_ext_emitters.hpp (hunk 7 above)
Comment (eshoguli) [reply src/plugins/intel_cpu/src/emitters/jit_mkldnn_ext_emitters.hpp]: done, thanks

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**